vulkan : update ggml_vk_instance_validation_ext_available

danbev opened the pull request:
This commit updates ggml_vk_instance_validation_ext_available() to check for VK_EXT_validation_features instead of
VK_KHR_portability_enumeration.

Based on how the returned boolean is used later in the code (to enable both the validation layer and the VK_EXT_validation_features extension), it appears the function may have been intended to check for the validation layer features extension. 

----

This is my first look at the Vulkan backend and I might be way off here...but the thing that caught my attention was the similarity between these functions [ggml_vk_instance_validation_ext_available](https://github.com/ggml-org/llama.cpp/blob/81017865ee444cf49ce0136f2be1e41a0270ff91/ggml/src/ggml-vulkan/ggml-vulkan.cpp#L12001-L12017) and
 [ggml_vk_instance_portability_enumeration_ext_available](https://github.com/ggml-org/llama.cpp/blob/81017865ee444cf49ce0136f2be1e41a0270ff91/ggml/src/ggml-vulkan/ggml-vulkan.cpp#L12018-L12034).

danbev committed:
vulkan : update ggml_vk_instance_validation_ext_available

This commit updates ggml_vk_instance_validation_ext_available() to
check for VK_EXT_validation_features instead of
VK_KHR_portability_enumeration.

Based on how the returned boolean is used later in the code (to enable
both the validation layer and the VK_EXT_validation_features extension),
it appears the function may have been intended to check for the
validation layer features extension.

0cc4m commented on ggml/src/ggml-vulkan/ggml-vulkan.cpp:
```diff
@@ -11998,22 +11998,25 @@ ggml_backend_reg_t ggml_backend_vk_reg() {
-    }
+    } catch (...) { /* It's alright if the layer is not present */ }
+
+    std::cerr << "ggml_vulkan: WARNING: Instance extension VK_EXT_validation_features not found." << std::endl;
```
```suggestion
    std::cerr << "ggml_vulkan: WARNING: Validation layer or layer extension VK_EXT_validation_features not found." << std::endl;
```

0cc4m reviewed (commented):
I... don't know how this happened. I guess I got distracted and just ran a test at some point and it worked as expected, so I never looked at this code again.

danbev commented on ggml/src/ggml-vulkan/ggml-vulkan.cpp:
```diff
@@ -11998,22 +11998,25 @@ ggml_backend_reg_t ggml_backend_vk_reg() {
-    if (!portability_enumeration_ext) {
-        std::cerr << "ggml_vulkan: WARNING: Instance extension VK_KHR_portability_enumeration not found." << std::endl;
-    }
+    } catch (...) { /* It's alright if the layer is not present */ }
```
Sorry about that, it was a left over from a previous iteration where I queries for a specific layer/extension. I'll remove this shortly. Thanks!

danbev committed:
remove try/catch

This was a left over from a previous iteration where I was explicitly
quering for a specific validation layer first, which would throw.

danbev committed:
update warning message about validation layers

0cc4m reviewed (approved):
Thank you for the fix!
